### ISSUE TITLE
[crash] panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by sungwoncho** [dgraph-io/dgraph#1164]
**Version**

```
Dgraph version   : v0.7.7
Commit SHA-1     : 25b5e99
Commit timestamp : 2017-07-10 13:56:17 +1000
Branch           : HEAD
```

**Reproduction**

1. Load data

[dgraph-1-2017-07-10-17-00.rdf.gz](https://github.com/dgraph-io/dgraph/files/1134625/dgraph-1-2017-07-10-17-00.rdf.gz)
[dgraph-schema-1-2017-07-10-17-00.rdf.gz](https://github.com/dgraph-io/dgraph/files/1134626/dgraph-schema-1-2017-07-10-17-00.rdf.gz)

2. Run the following query

```
{
  var(func: anyoftext(Text, "movie")) @cascade {
  	p as Post {
      ~Post(orderdesc: Timestamp, first: 1)@filter(anyoftext(Text, "movie")) {
      	
      }
    }
  }
    
  posts(func: var(p), first: 1) {
    Title {
      Text
    }
    Body {
      Text
    }
  }
}
```

3. Dgraph crashes with the following error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x465791f]

goroutine 17718 [running]:
github.com/dgraph-io/dgraph/algo.IntersectWith(0xc44bbb8c60, 0x0, 0xc44bbb8c60)
        /Users/travis/gopath/src/github.com/dgraph-io/dgraph/algo/uidlist.go:44 +0x2f
github.com/dgraph-io/dgraph/query.ProcessGraph(0x5012940, 0xc4425da540, 0xc44247f8c0, 0x0, 0xc44b12a900)
        /Users/travis/gopath/src/github.com/dgraph-io/dgraph/query/query.go:1620 +0x2c41
created by github.com/dgraph-io/dgraph/query.ProcessQuery
        /Users/travis/gopath/src/github.com/dgraph-io/dgraph/query/query.go:1254 +0x729
```

**Comment by manishrjain** [comment]
Try it on master as well.

**Comment by sungwoncho** [comment]
Will try later. Currently not able to build `master`.

```
# github.com/dgraph-io/dgraph/worker
../../worker/worker.go:63: undefined: grpc.MaxRecvMsgSize
../../worker/worker.go:64: undefined: grpc.MaxSendMsgSize
```

Nightly build exits with 

```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/dgraph-io/badger/table.(*Table).read(0xc424cce000, 0xffffffff4c0ed61f, 0xb741e288, 0xa50b8a7, 0x4, 0x4, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/dgraph-io/badger/table/table.go:200 +0x10e
github.com/dgraph-io/badger/table.(*Table).readNoFail(0xc424cce000, 0xffffffff4c0ed61f, 0xb741e288, 0xa50b8a7, 0x4, 0x4)
        /Users/travis/gopath/src/github.com/dgraph-io/badger/table/table.go:212 +0x43
github.com/dgraph-io/badger/table.(*Table).readIndex(0xc424cce000, 0x7000000, 0x7000000)
        /Users/travis/gopath/src/github.com/dgraph-io/badger/table/table.go:223 +0xa6
github.com/dgraph-io/badger/table.OpenTable(0xc4200982e8, 0x1, 0x0, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/dgraph-io/badger/table/table.go:149 +0x162
github.com/dgraph-io/badger.newLevelsController(0xc42019e600, 0xc4202e2ab0, 0x4924e60, 0x2)
        /Users/travis/gopath/src/github.com/dgraph-io/badger/levels.go:102 +0x566
github.com/dgraph-io/badger.NewKV(0xc4200b7d90, 0xc4200b7d90, 0x2d, 0x0)
        /Users/travis/gopath/src/github.com/dgraph-io/badger/kv.go:160 +0x5b3
main.main()
        /Users/travis/gopath/src/github.com/dgraph-io/dgraph/cmd/dgraph/main.go:791 +0x23a
```

**Comment by manishrjain** [comment]
I meant, for @ashwin95r : Befor eyou write a fix, try to replicate this on master as well.

**Comment by ashwin95r** [comment]
Doesn't crash on master. Will fix in release.

**Comment by ashwin95r** [comment]
Fixed on release/v0.7.7
